extensions/cdc-mysql/src/test/java/com/hazelcast/jet/cdc/mysql/MySqlCdcNetworkIntegrationTest.java

```diff
@@ -0,0 +1,410 @@
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.StreamSource;
+import com.hazelcast.jet.retry.RetryStrategies;
+import com.hazelcast.jet.retry.RetryStrategy;
+import com.hazelcast.jet.test.SerialTest;
+import com.hazelcast.test.HazelcastSerialParametersRunnerFactory;
+import com.hazelcast.test.annotation.NightlyTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.jet.Util.entry;
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.testcontainers.containers.MySQLContainer.MYSQL_PORT;
+
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
+@Category({SerialTest.class, NightlyTest.class})
+public class MySqlCdcNetworkIntegrationTest extends AbstractCdcIntegrationTest {
+
+    private static final long RECONNECT_INTERVAL_MS = SECONDS.toMillis(1);
+
+    @Parameter(value = 0)
+    public RetryStrategy reconnectBehavior;
+
+    @Parameter(value = 1)
+    public boolean resetStateOnReconnect;
+
+    @Parameter(value = 2)
+    public String testName;
+
+    @Parameters(name = "{2}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { RetryStrategies.never(), false, "fail"},
+                { RetryStrategies.indefinitely(RECONNECT_INTERVAL_MS), false, "reconnect"},
+                { RetryStrategies.indefinitely(RECONNECT_INTERVAL_MS), true, "reconnect w/ state reset"}
+        });
+    }
+
+    @Test
+    public void when_noDatabaseToConnectTo() throws Exception {
+        MySQLContainer<?> mysql = initMySql(null, 0);
+        String containerIpAddress = mysql.getContainerIpAddress();
+        stopContainer(mysql);
+
+        int port = findRandomOpenPortInRange(MYSQL_PORT + 100, MYSQL_PORT + 1000);
```

Comment:
Do we actually need `findRandomOpenPortInRange` method? It seems `findRandomOpenPort()` could be ok.

Reply:
My reasoning was to use this ranged method in situations when we don't immediately occupy the free port we detect. Maybe in these situations, being on a machine with lots of activity, the port gets occupied before we attempt to do so ourselves. Whereas if you use a specific range the likelihood of this happening seems smaller. I might be completely off though, this might not match how the OS actually allocates ports. On the other hand: do we really gain anything by removing it?

Reply:
Let's keep it as it is. You're right that we do not gain anything by removing it.